### PR TITLE
SYN-4929: devops guide for tls:ca:dir setting

### PR DESCRIPTION
**Pull request by MichaelSquires**

### CODE REVIEW

**Comment by MichaelSquires** [comment]
[devopsguide.tar.gz](https://github.com/user-attachments/files/16910744/devopsguide.tar.gz)

File: docs/synapse/devopsguide.rst [hunk 1]
```diff
@@ -981,6 +981,123 @@ this to add a user named ``foouser`` with the uid 1234::
...
+    services:
+      00.cortex:
+        user: "999"
+        image: vertexproject/synapse-cortex:v2.x.x
+        network_mode: host
+        restart: unless-stopped
+        volumes:
+            - ./storage:/vertex/storage
+        environment:
+            SYN_CORTEX_TLS_CA_DIR: /vertex/storage/tls-ca-certs
```
Comment (vEpiphyte) [inline docs/synapse/devopsguide.rst:1005]: Can we show mapping in a separate volume?
In some production cases, mapping multiple volumes under `/vertex/storage` may not be possible; so we may not want to imply that is the best way to do this configuration.

File: docs/synapse/devopsguide.rst [hunk 2]
```diff
@@ -981,6 +981,123 @@ this to add a user named ``foouser`` with the uid 1234::
...
+#. Create a config map from your CA certificate files.
+
+    The example has a pre-created root CA and an intermediate CA certificate in PEM format::
+
+        $ ls -l ./cas
+        total 8
+        -rw-rw-r-- 1 user user 1708 Feb 14 19:19 intermediate.crt
+        -rw-rw-r-- 1 user user 1696 Feb 14 19:19 root.crt
+
+        $ kubectl create configmap zscalar-ca-certs --from-file ./cas
```
Comment (vEpiphyte) [inline docs/synapse/devopsguide.rst:1027]: Lets call this configmap something generic such as `tls-ca-certs`.

Comment (MichaelSquires) [reply docs/synapse/devopsguide.rst:1027]: Thanks, I meant to update that and missed it

File: docs/synapse/devopsguide.rst [hunk 3]
```diff
@@ -981,6 +981,123 @@ this to add a user named ``foouser`` with the uid 1234::
...
+    the following will be seen::
+
+        inet:http:resp: {'code': -1, 'reason': "Exception occurred during request: ClientConnectorCertificateError ...", ...}
+
+In this example, the volume with the configmap contains symlinks which are
+treated as directories. When the SSLContext is created with those additional
+files, Synapse attempts to load the directory, will fail and then log an
+exception. This does not stop the SSLContext from being created with the
+additonal CA files, but does generate a lot of log messages. This will be
+addressed in a future Synapse version.
```
[the remark this answers is not in the document]
Comment (MichaelSquires) [reply docs/synapse/devopsguide.rst:1094]: No idea. That was from the text you pasted into the ticket. I'll check though. :laughing:

[the remark this answers is not in the document]
Comment (MichaelSquires) [reply docs/synapse/devopsguide.rst:1094]: SYN-7973

File: docs/synapse/devopsguide.rst [hunk 4]
```diff
@@ -981,6 +981,124 @@ this to add a user named ``foouser`` with the uid 1234::
     Successfully built 21a12f395462
     Successfully tagged vertexproject/synapse-aha:v2.113.0-foouser
 
+Configure Custom CA Certificates for HTTP Requests
+--------------------------------------------------
+
+The Cortex can be configured to use additional CA certificates when making HTTP
```
Comment (vEpiphyte) [inline docs/synapse/devopsguide.rst:987]:
```
The Cortex and Axon can be configured to use additional CA certificates when making HTTP
```

File: docs/synapse/devopsguide.rst [hunk 5]
```diff
@@ -981,6 +981,124 @@ this to add a user named ``foouser`` with the uid 1234::
...
+requests via Storm. To do this, you need to provide the certificates (in DER or
+PEM format) to the Cortex in a directory on disk, and then configure the Cortex
+to look up that directory via the ``tls:ca:dir`` configuration option.
+
+The following Compose file shows an example using this option.
```
Comment (vEpiphyte) [inline docs/synapse/devopsguide.rst:992]:
```
 this option with the Cortex.
```

File: docs/synapse/devopsguide.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (vEpiphyte) [reply docs/synapse/devopsguide.rst:1094]: Logspam was fixed in https://github.com/vertexproject/synapse/pull/3179/files

File: docs/synapse/devopsguide.rst [hunk 6]
```diff
@@ -981,6 +981,124 @@ this to add a user named ``foouser`` with the uid 1234::
     Successfully built 21a12f395462
     Successfully tagged vertexproject/synapse-aha:v2.113.0-foouser
 
+Configure Custom CA Certificates for HTTP Requests
+--------------------------------------------------
+
+The Cortex can be configured to use additional CA certificates when making HTTP
+requests via Storm. To do this, you need to provide the certificates (in DER or
```
Comment (vEpiphyte) [inline docs/synapse/devopsguide.rst:988]: Have you tested with DER format? I've only tested with PEM format and the sslcontext docs note it is PEM encoded: https://docs.python.org/3/library/ssl.html#ssl.SSLContext.load_verify_locations

Comment (MichaelSquires) [reply docs/synapse/devopsguide.rst]: Good call. DER does not work. Updated text.

**Review by invisig0th — APPROVED** [review]

**Pull request merged**